MANIFEST.in

```diff
@@ -1,3 +1,4 @@
-include LICENSE
-include README.md
-include requirements.txt
+global-exclude *.png *.gif
```

Comment:
You need to be careful about excluding every image possible because it may later cause problems with images included in the Sphinx docs. Downstream packagers usually build those docs and the missing images will cause warnings turned into errors. Maybe list the entries explicitly?

Reply:
You are correct.  Though for images I always prefer using a URL instead of referencing file in the repo.  With this, changes to media files at these URLs will be reflected everywhere, no matter where the package might be.  E.g. https://pypi.org/project/proxy.py/ image here is now updated even without a push.

I intend to follow the same for docs.  Better to use CDN urls, instead of putting images in the repo.  On that note, I can very well just delete `png` and `gif` from the repo, as they are referenced as URL everywhere.

I feel there shouldn't be any media files in the repo.  There is a `adblock.json` which can be removed as well, I skipped removing it for now as `sdist` users can still use that file as a reference for running the plugin.

Wdyt?

Reply:
In future, if we include any media in repo which must be reference from within the repo, then contributors can explicitly whitelist that here?   At-least, we know, no unnecessary images are going into the package.

Reply:
> You are correct. Though for images I always prefer using a URL instead of referencing file in the repo. With this, changes to media files at these URLs will be reflected everywhere, no matter where the package might be. E.g. [pypi.org/project/proxy.py](https://pypi.org/project/proxy.py/) image here is now updated even without a push.
> 
> I intend to follow the same for docs. Better to use CDN urls, instead of putting images in the repo. On that note, I can very well just delete `png` and `gif` from the repo, as they are referenced as URL everywhere.

This is why I was talking about the project's future. For images that need to be included on PyPI you don't have a choice but not for the Sphinx docs. Some images may be part of the docs. These docs should be buildable offline. Besides, on different commits, you may have different content (including images). Imagine that you've replaced the image in the latest commit and it suddenly gets replaced in the docs for a very ancient version 2.0.0 displaying something that doesn't make sense for that version.

> I feel there shouldn't be any media files in the repo. There is a `adblock.json` which can be removed as well, I skipped removing it for now as `sdist` users can still use that file as a reference for running the plugin.
> 
> Wdyt?

This statement is not universal, there are cases where it's necessary.

Reply:
> Imagine that you've replaced the image in the latest commit and it suddenly gets replaced in the docs for a very ancient version 2.0.0

Totally.  And that is part of the *(product)* strategy :).  Hero image is more of an advertising material.  It must always reflect the state-of-the-art form of the product.  This will also compel users to click the image and visit the GH repo :).  No matter wherever they see the hero image.

> This statement is not universal, there are cases where it's necessary.

Agreed.  For now I feel let's keep it explicit, until we start to run into issue.  With implicit, `sdist` size can again go up without anyone noticing.  It was nice that you noticed, could have taken time before I realize it myself.  May-be, we should setup a CI failure if `sdist` size goes over an expected limit.  Example, `ProxyPy.png` and `shortlink.gif` together accounted for `~1.5Mb` which is `~10x` the actual contents of `sdist`.

Reply:
Ok, as I was in `MANIFEST.in`, I also made this change.  Let's take this risk for now, not worth denying your request ;)